### PR TITLE
Collection: add a test for multiple credentials on a jt

### PR DESCRIPTION
**Pull request by squidboylan**

### CODE REVIEW

File: awx_collection/tests/integration/targets/tower_job_template/tasks/main.yml [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+    jt1: "hello-world {{ lookup('password', '/dev/null chars=ascii_letters length=16') }}"
+    jt2: "hello-world {{ lookup('password', '/dev/null chars=ascii_letters length=16') }}"
+
+- name: Create SCM Credential1
+  tower_credential:
+    name: "{{ cred1 }}"
+    organization: Default
+    kind: tower
+
+- name: Create SCM Credential2
```
Comment (AlanCoding) [inline awx_collection/tests/integration/targets/tower_job_template/tasks/main.yml]: not SCM cred

File: awx_collection/tests/integration/targets/tower_job_template/tasks/main.yml [hunk 2]
```diff
@@ -15,17 +35,83 @@
...
+    credentials: ["{{ cred1 }}", "{{ cred2 }}"]
+    job_type: run
+    state: present
+  register: result
+
+- assert:
+    that:
+      - "result is changed"
+
+# This doesnt work if you include the credentials parameter
```
Comment (AlanCoding) [inline awx_collection/tests/integration/targets/tower_job_template/tasks/main.yml:53]: oh... should it not work? It seems like it should

Comment (AlanCoding) [reply awx_collection/tests/integration/targets/tower_job_template/tasks/main.yml:53]: b/c people parameterize their playbooks with the state parameter

Comment (squidboylan) [reply awx_collection/tests/integration/targets/tower_job_template/tasks/main.yml:53]: ahh yes, then we should fix this, the error message i see is:

```
TASK [tower_job_template : Delete Job Template 1] ****************************************************************************************************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'id'
fatal: [testhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/home/squid/.ansible/tmp/ansible-tmp-1584127597.5142252-20525162028967/AnsiballZ_tower_job_template.py\", line 116, in <module>\n    _ansiballz_main()\n  File \"/home/squid/.ansible/
tmp/ansible-tmp-1584127597.5142252-20525162028967/AnsiballZ_tower_job_template.py\", line 108, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/squid/.ansible/tmp/ansible-tmp-1584127597.5142252-20525162028967/AnsiballZ_tower_job_template.py\", line
 54, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.awx.awx.plugins.modules.tower_job_template', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_n
ame, mod_spec)\n  File \"/usr/lib/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_tower_job_template_payload_hwp07zj9/ansible_to
wer_job_template_payload.zip/ansible_collections/awx/awx/plugins/modules/tower_job_template.py\", line 421, in <module>\n  File \"/tmp/ansible_tower_job_template_payload_hwp07zj9/ansible_tower_job_template_payload.zip/ansible_collections/awx/awx/plugins/modules/tower_job_template.py\", line 41
0, in main\nKeyError: 'id'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

**Review by beeankha — APPROVED** [review]

**Comment by AlanCoding** [comment]
are these failing currently and need a patch?

**Comment by squidboylan** [comment]
> are these failing currently and need a patch?

I just updated it and it should work

**Review by AlanCoding — APPROVED** [review]

**Comment by squidboylan** [comment]
We will fix the bug of `state: 'absent'` in a future PR, we have things in flight which may fix it as a side effect I think

**Pull request merged**